rate_limit.py

```diff
@@ -0,0 +1,98 @@
+"""
+This module contains functions for rate limiting requests.
+"""
+
+import datetime
+import signal
+import sys
+from uuid import uuid4
+
+from apscheduler.schedulers import background
+import gradio as gr
+
+
+class InvalidTokenException(Exception):
+  pass
+
+
+class UserRateLimitException(Exception):
+  pass
+
+
+class SystemRateLimitException(Exception):
+  pass
+
+
+class RateLimiter:
+
+  def __init__(self, daily_limit=10000):
+    self.requests = {}
+    self.request_count = 0
+    self.daily_limit = daily_limit
+    self.scheduler = background.BackgroundScheduler()
```

Comment:
I think this scheduler is no longer used?

Reply:
yes thanks!